library/src-3.x/dotty/internal/StringContextMacro.scala

```diff
@@ -0,0 +1,777 @@
+package dotty.internal
+
+import scala.quoted._
+import scala.quoted.matching._
+import scala.tasty.Reflection
+import scala.language.implicitConversions
+import scala.quoted.Exprs.LiftedExpr
+import reflect._
+
+object StringContextMacro {
+
+  /** Implemetation of scala.StringContext.f used in Dotty while the standard library is still not bootstrapped */
+  inline def f(sc: => StringContext)(args: Any*): String = ${ fImpl('sc, 'args) }
+
+  /** This trait defines a tool to report errors/warnings that do not depend on Position. */
+  trait Reporter{
+
+    /** Reports error/warning of size 1 linked with a part of the StringContext.
+      *
+      * @param message the message to report as error/warning
+      * @param index the index of the part inside the list of parts of the StringContext
+      * @param offset the index in the part String where the error is
+      * @return an error/warning depending on the function
+      */
+    def partError(message : String, index : Int, offset : Int) : Unit
+    def partWarning(message : String, index : Int, offset : Int) : Unit
+
+    /** Reports error linked with an argument to format.
+      *
+      * @param message the message to report as error/warning
+      * @param index the index of the argument inside the list of arguments of the format function
+      * @return an error depending on the function
+      */
+    def argError(message : String, index : Int) : Unit
+
+    /** Reports error linked with the list of arguments or the StringContext.
+      *
+      * @param message the message to report in the error
+      * @return an error
+      */
+    def strCtxError(message : String) : Unit
+    def argsError(message : String) : Unit
+
+    /** Claims whether an error or a warning has been reported
+      *
+      * @return true if an error/warning has been reported, false
+      */
+    def hasReported() : Boolean
+
+    /** Stores the old value of the reported and reset it to false */
+    def resetReported() : Unit
+
+    /** Restores the value of the reported boolean that has been reset */
+    def restoreReported() : Unit
+  }
+
+  /** Retrieves a String from an Expr containing it
+    *
+    * @param expression the Expr containing the String
+    * @return the String contained in the given Expr
+    * quotes an error if the given Expr does not contain a String
+    */
+  private def literalToString(expression : Expr[String])(implicit reflect: Reflection) : String = expression match {
+    case Const(string : String) => string
+    case _ => QuoteError("Expected statically known part list")
```

Comment:
I suppose this should be 
```scala
QuoteError("Expected statically known part list", expression)
```
to provide a more precise position

Reply:
Though will never happen if we write `f"..."`, only if we write `StringContext(...).f(...)`